pkg/pillar/zboot/zboot_linux.go

```diff
@@ -20,3 +20,13 @@ func zbootMount(devname string, target string, fstype string,
 	}
 	return syscall.Mount(devname, target, fstype, flagsLinux, data)
 }
+
+func zbootUnmount(target string, force bool) (err error) {
+	var flagsLinux int
+	if force {
+		flagsLinux |= syscall.MNT_FORCE
+	} else {
+		flagsLinux |= syscall.MNT_DETACH
+	}
```

Comment:
As I can see MNT_FORCE not supported for squashfs: https://man7.org/linux/man-pages/man2/umount.2.html
(I dd our rootfs onto /dev/sdb5 and run `sudo chroot /tmp/tst/ /bin/sh` in another session after mount)
```
$ mount|grep /tmp/tst
/dev/sdb5 on /tmp/tst type squashfs (ro,relatime,errors=continue)
$ sudo umount --force /tmp/tst
umount: /tmp/tst: target is busy.
```
On the other hand, lazy unmount (MNT_DETACH) will give us wrong impression that the target was unmounted and we will hit the error during mount:
```
$ sudo mount /dev/sdb5 /tmp/tst
$ sudo umount --lazy /tmp/tst
$ sudo mount /dev/sdb5 /tmp/tst2
mount: /tmp/tst2: /dev/sdb5 already mounted or mount point busy.
```
Do you have any reproducer to check the problem? It looks quite strange that we hit the problem with unmount but have no problems with subsequent mount.
Probably it would be better to use Fadvise/Madvise to avoid cache usage.

Reply:
@giggsoff we see this when running ztests on the PRs, but I don't see it anywhere else were I can reproduce it.
Based on your manual tests I'll just to do sleep loop (maybe for 10 times each sleeping one second instead of just three), and then we can see if that makes the issue go away.

Reply:
Alternatively Petr can add sleep between his commands and see how much amount of sleep is required. We can approximate to that value in the code ?